schema/query/query.go

```diff
@@ -0,0 +1,383 @@
+package query
+
+import (
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/rs/rest-layer/schema"
+)
+
+const (
+	cmdAnd           = "$and"
+	cmdOr            = "$or"
+	cmdExists        = "$exists"
+	cmdIn            = "$in"
+	cmdNotIn         = "$nin"
+	cmdNotEqual      = "$ne"
+	cmdLowerThan     = "$lt"
+	cmdLowerOrEqual  = "$lte"
+	cmdGeaterThan    = "$gt"
+	cmdGeaterOrEqual = "$gte"
+	cmdRegex         = "$regex"
```

Comment:
Since this names are internal, it doesn't matter to much, but it's probably better to prefix them 'op' for operator rather than cmd for command;

https://docs.mongodb.com/manual/reference/operator/query/